### PR TITLE
remote repository, svn resource, path normalizer api

### PR DESCRIPTION
**Pull request by Yanpas**

### CODE REVIEW

**Comment by JohnstonCode** [comment]
Can you also explain this?

**Comment by Yanpas** [comment]
History lens supports viewing history of not only local repositories, but any svn url too. So I've decided to separate this functionality to separate class called RemoteRepository and interface. The only things you can do with it is to see log and get files under some revisions.

PathNormalizer is for converting local paths to remote ones and vice versa. It returns svn resource identifiers. Since the repository may be remote localFullPath may return undefined.

**Comment by JohnstonCode** [comment]
Sorry for being so absent being Ill/busy (Christmas stuff)

File: src/svnRI.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import * as path from "path";
```
[the remark this answers is not in the document]
Comment (Yanpas) [reply src/svnRI.ts]: same as URI, Svn Resource Identifier

**Pull request merged**